app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -607,31 +616,34 @@ void OpenXRInputSource::EmulateControllerFromHand(device::RenderMode renderMode,
     }
 
     // Check whether palm of left hand is facing the head, to show navigate-back button
-    bool leftPalmFacesHead = false;
-    if (mHandeness == OpenXRHandFlags::Left) {
-#if defined(OCULUSVR)
-        if (IsHandJointPositionValid(XR_HAND_JOINT_PALM_EXT))
-            leftPalmFacesHead = !mHasAimState;
-#else
-        XrHandJointEXT palmJointIndex = XR_HAND_JOINT_PALM_EXT;
-#if defined(PICOXR)
-        // Pico runtime doesn't provide palm joint info, so we use the wrist instead.
-        palmJointIndex = XR_HAND_JOINT_WRIST_EXT;
+    bool palmFacesHead = false;
+    XrHandJointEXT palmJointIndex = GetHandJointIndexForAim();
+    bool systemTakesOverWhenHandsFacingHead = false;
+#if defined(OCULUS)
+    systemTakesOverWhenHandsFacingHead = true;
 #endif
-        if (IsHandJointPositionValid(palmJointIndex)) {
-            vrb::Matrix palmMatrix = jointTransforms[palmJointIndex];
-            // For the hand we take the Y axis because that corresponds to head's Z axis when
-            // the hand is in upright position facing head (the gesture we want to detect).
+    if (IsHandJointPositionValid(palmJointIndex)) {
```

Comment:
> Technically the palm joint doesn't need to be valid for Oculus to show the back-nav icon (assuming we forced `mSupportsFBHandTrackingAim` to false). I know it is stretched but I think this change will not add much consistency to the Oculus case.

That's a good point I'll change that?

Reply:
> My point is that I don't see practical any scenario where we disable FB aim extension on Oculus. For example, I just tested the patch forcing `mSupportsFBHandTrackingAim` to false, and I could see the aim state (beam visible) together with the Oculus system's back nav icon. In that state, user would expect to be able to use the nav-back gesture, because the icon is visible, however it does nothing because the aim is still active.

Really? The patch addresses that case too, I see how the ray is not painted in Oculus forcing the FB extension off.

Reply:
> > Technically the palm joint doesn't need to be valid for Oculus to show the back-nav icon (assuming we forced `mSupportsFBHandTrackingAim` to false). I know it is stretched but I think this change will not add much consistency to the Oculus case.
> 
> Note that condition was already present in the current code.

Right.

Reply:
> > My point is that I don't see practical any scenario where we disable FB aim extension on Oculus. For example, I just tested the patch forcing `mSupportsFBHandTrackingAim` to false, and I could see the aim state (beam visible) together with the Oculus system's back nav icon. In that state, user would expect to be able to use the nav-back gesture, because the icon is visible, however it does nothing because the aim is still active.
> 
> Really? The patch addresses that case too, I see how the ray is not painted in Oculus forcing the FB extension off.

Well with some trying I could make it to see both the ray and the oculus back button, and in that case the pinch doesn't do back-nav. Bu it is true that it is unlikely to be a problem since it is a very narrow angle that allows that.